File: Greedy/605.Can_Place_Flowers/Can_Place_Flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        flowerbed.insert(flowerbed.begin(), 0); flowerbed.push_back(0);
        int ans = 0;
        for(int i = 1; i < size(flowerbed) - 1; i++){
            if (!flowerbed[i - 1] && !flowerbed[i] && !flowerbed[i + 1]){ 
                flowerbed[i] = 1;
                ans += 1;
            }
        }
        return ans >= n;
    }
};

File: Greedy/605.Can_Place_Flowers/Can_Place_Flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        flowerbed.insert(0, 0); flowerbed.append(0)
        ans = 0
        for i in range(1, len(flowerbed) - 1):
            if not(flowerbed[i - 1]) and not(flowerbed[i]) and not(flowerbed[i + 1]): 
                flowerbed[i] = 1
                ans += 1
        return ans >= n